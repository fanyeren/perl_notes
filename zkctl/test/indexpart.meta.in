#!/bin/bash
source $(dirname $0)/common
TYPE=index-partition
VALUE=dnews_2
raw='"group":"group_1","index_layer":"layer1","index_type":"dnews","replication_factor":"1"'

#准备环境
node_clear $TYPE
node_clear index-type
./zkctl -w index-type add --index-type-id=dnews --data={}

./zkctl -w $TYPE add --$TYPE-id=$VALUE --data='{'$raw'}'
./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
node_clear $TYPE
./zkctl -w $TYPE add --$TYPE-id=$VALUE --data={}
./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE add-meta-attr --$TYPE-id=$VALUE --key=${kv[0]} --value=${kv[1]}
    echo "Add meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -w $TYPE get-meta-attr --$TYPE-id=$VALUE --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -w $TYPE set-meta-attr --$TYPE-id=$VALUE --key=${kv[0]} --value='hjz'
    echo "Set meta ${kv[0]}=${kv[1]} result $?"
    ./zkctl -w $TYPE get-meta-attr --$TYPE-id=$VALUE --key=${kv[0]}
    echo "Get meta ${kv[0]}=${kv[1]} result $?"
done

./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
echo "got all $TYPE result result=$?"

for item in `echo $raw|tr -s "," " "`; do
    rawkv=`echo $item|tr -d \" | tr -s ":" " "`
    kv=($rawkv)
    ./zkctl -w $TYPE del-meta-attr --$TYPE-id=$VALUE --key=${kv[0]}
    echo "del meta attr ${kv[0]} result=$?"
done

./zkctl -w $TYPE get-meta-all --$TYPE-id=$VALUE
echo "got empty $TYPE result result=$?"

./zkctl -w $TYPE del --$TYPE-id=$VALUE
node_clear index-type
