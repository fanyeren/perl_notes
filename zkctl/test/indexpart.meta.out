{"group":"group_1","index_layer":"layer1","index_type":"dnews","replication_factor":"1"}
{}
Add meta group=group_1 result 0
group_1
Get meta group=group_1 result 0
Set meta group=group_1 result 0
hjz
Get meta group=group_1 result 0
Add meta index_layer=layer1 result 0
layer1
Get meta index_layer=layer1 result 0
Set meta index_layer=layer1 result 0
hjz
Get meta index_layer=layer1 result 0
Add meta index_type=dnews result 0
dnews
Get meta index_type=dnews result 0
Set meta index_type=dnews result 0
hjz
Get meta index_type=dnews result 0
Add meta replication_factor=1 result 0
1
Get meta replication_factor=1 result 0
Set meta replication_factor=1 result 0
hjz
Get meta replication_factor=1 result 0
{"group":"hjz","index_layer":"hjz","index_type":"hjz","replication_factor":"hjz"}
got all index-partition result result=0
del meta attr group result=0
del meta attr index_layer result=0
del meta attr index_type result=0
del meta attr replication_factor result=0
{}
got empty index-partition result result=0
